/*
 * This file was automatically generated by EvoSuite
 * Thu May 29 00:34:48 GMT 2025
 */

package original;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import original.Solution2503;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Solution2503_ESTest extends Solution2503_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution2503 solution2503_0 = new Solution2503();
      int[][] intArray0 = new int[5][6];
      int[] intArray1 = new int[9];
      intArray1[0] = (-4569);
      intArray1[1] = 1484;
      intArray0[0] = intArray1;
      int[] intArray2 = solution2503_0.maxPoints(intArray0, intArray0[1]);
      assertArrayEquals(new int[] {1, 1, 1, 1, 1, 1}, intArray2);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution2503 solution2503_0 = new Solution2503();
      int[][] intArray0 = new int[6][9];
      int[] intArray1 = new int[0];
      intArray0[1] = intArray1;
      int[] intArray2 = solution2503_0.maxPoints(intArray0, intArray0[1]);
      assertEquals(0, intArray2.length);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution2503 solution2503_0 = new Solution2503();
      int[][] intArray0 = new int[0][6];
      int[] intArray1 = new int[3];
      // Undeclared exception!
      try { 
        solution2503_0.maxPoints(intArray0, intArray1);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index 0 out of bounds for length 0
         //
         verifyException("original.Solution2503", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution2503 solution2503_0 = new Solution2503();
      int[][] intArray0 = new int[4][4];
      int[] intArray1 = new int[6];
      intArray1[0] = 5;
      int[] intArray2 = solution2503_0.maxPoints(intArray0, intArray1);
      assertArrayEquals(new int[] {16, 0, 0, 0, 0, 0}, intArray2);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Solution2503 solution2503_0 = new Solution2503();
      int[] intArray0 = new int[1];
      // Undeclared exception!
      try { 
        solution2503_0.maxPoints((int[][]) null, intArray0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("original.Solution2503", e);
      }
  }
}
